### PR TITLE
Refines Principal Software Engineer role definition

### PR DESCRIPTION
**Pull request by andrewabest**
I've made some updates to the PSE role definition to:

1 - align it with some things that are now relevant and important to Octopus, like the technical strategy
2 - ensure it reflects my personal experiences in the role and what I consider important aspects of PSE at Octopus
3 - reinforce some of the cultural aspects of the role

I'll await feedback from our current PSE's prior to merging.

### CODE REVIEW

**Review by droyad — COMMENTED** [review]
LGTM

File: Software-Engineering/L5-Principal-Software-Engineer.md [hunk 1]
```diff
@@ -60,17 +70,19 @@
 - I collaborated with other senior leaders to ensure I was aware of all major initiatives at Octopus and could account for them in my own initiatives.
 - I identified a significant problem/opportunity and created a lasting best-fit solution that aligned with our strategy.
 - Instead of simply patching or extending an existing solution that was not fit for purpose anymore, my contribution opened up a whole new area of strategic possibility.
+- I participate in our code review and interview processes, and provide feedback as to how they can be improved regularly.
```
Comment (Waldo000000) [inline Software-Engineering/L5-Principal-Software-Engineer.md:73]: [nitpick] Prefer past-tense for concrete `Examples`:
```suggestion
- I regularly participated in our code review and interview processes and provided feedback on how they could be improved.
```

File: Software-Engineering/L5-Principal-Software-Engineer.md [hunk 2]
```diff
@@ -39,18 +45,22 @@
...
 
 </details>
 
 ## 🧭 Culture and Leadership
 
+- I model our cultural safety by thinking in the open, asking imperfect questions, being wrong, and accepting corrections.
+- I generate energy within the teams I work with, motivating them to achieve great outcomes.
 - I facilitate effective and timely decision making, especially when the decision affects multiple teams.
-- I build tools/frameworks/guidance that improves productivity for many people/teams.
+- I build solutions that improve the productivity and quality of many people/teams.
```
Comment (Waldo000000) [inline Software-Engineering/L5-Principal-Software-Engineer.md:61]:
```suggestion
- I build solutions that improve the productivity and/or quality of work of many people/teams.
```

File: Software-Engineering/L5-Principal-Software-Engineer.md [hunk 3]
```diff
@@ -10,27 +10,33 @@
...
 
 - I successfully completed complex tasks spanning multiple domains and teams with high impact. 
-- I helped a team reduce complexity through sound architectural thinking.
-- I identified a problem that was timely, important, and impactful, and worked closely with product, engineering, and executive stakeholders to articulate and refine a solution.
-- I handled multiple conflicting priorities and sequenced work resulting in the best collective outcome.
+- I helped a team reduce complexity and risk through sound architectural thinking.
+- I developed a plan to evolve the architecture of one of the core components of Octopus within my impact radius.
+- I identified a problem that was timely, important, and impactful, and worked closely with product, engineering, and executive stakeholders to articulate and refine a solution and delivery plan.
+- I worked with teams to manage multiple conflicting priorities and helped sequence work resulting in the best collective outcome.
+- I ensured the feature I was helping a team plan made decisions that supported a feature I knew another team was planning by collaborating with that team's aligned principal engineer.
```
Comment (Waldo000000) [inline Software-Engineering/L5-Principal-Software-Engineer.md:28]: I found this sentence difficult to parse; here's a reword to consider:
```suggestion
- I collaborated with another principal engineer to ensure that a feature I was helping a team with would support a feature I knew another team was planning.
```

File: Software-Engineering/L5-Principal-Software-Engineer.md [hunk 4]
```diff
@@ -39,18 +45,22 @@
...
+- Engineers at Octopus seek out to work on my teams to learn from me.
+- After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager to uplift their skills in the identified growth area.
 - I helped a team make a course correction based on new information or ideas.
 - I provided appropriate technical freedom, and framing, that allowed the Lead Software Engineers on my teams to thrive independently.
 
 </details>
 
 ## 🧭 Culture and Leadership
 
+- I model our cultural safety by thinking in the open, asking imperfect questions, being wrong, and accepting corrections.
```
Comment (Waldo000000) [inline Software-Engineering/L5-Principal-Software-Engineer.md:58]: Terms like "model our cultural safety" and "imperfect questions" felt a bit abstract/open to interpretation. Here's a reword to consider, which puts the intended _outcome_ upfront:

```suggestion
- I create an environment where others feel safe, by modeling vulnerability, thinking in the open, being wrong, and accepting corrections.
```

**Review by Waldo000000 — APPROVED** [review]
Thanks Besty! ✨ 

1. I left some minor suggestions
2. Overall, these changes look good to me; they appear to clarify existing expectations rather than change the nature of existing expectations
3. I'm keen to also hear from other PEs (and their managers), so thanks for already reaching out to them for feedback
4. For context: This ad-hoc update would complement our regular review and update of this role ladder, which happens after calibration at each perf review cycle ([internal link](https://octopushq.atlassian.net/wiki/spaces/RND/pages/2236940315/Engineering+calibration+process#How-we-use-calibration-to-refine-our-role-ladders)).

**Comment by andrewabest** [comment]
Thanks for the suggestions @Waldo000000 👍 all good ones, and didn't change the content of what I'd added. Much appreciated!

File: Software-Engineering/L5-Principal-Software-Engineer.md [hunk 5]
```diff
@@ -39,18 +45,22 @@
...
+- Engineers at Octopus seek out to work on my teams to learn from me.
+- After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager to uplift their skills in the identified growth area.
 - I helped a team make a course correction based on new information or ideas.
 - I provided appropriate technical freedom, and framing, that allowed the Lead Software Engineers on my teams to thrive independently.
 
 </details>
 
 ## 🧭 Culture and Leadership
 
+- I create an environment where others feel safe, by modeling vulnerability, thinking in the open, being wrong, and accepting corrections.
```
Comment (TomPeters) [inline Software-Engineering/L5-Principal-Software-Engineer.md:58]: I like this one.

**Review by TomPeters — COMMENTED** [review]
All of these changes look reasonable to me. There's nothing here that surprises me, or doesn't align with my expectations or experiences.

File: Software-Engineering/L5-Principal-Software-Engineer.md (hunk 5 above)
Comment (matt88shepherd) [reply Software-Engineering/L5-Principal-Software-Engineer.md:58]: +1

File: Software-Engineering/L5-Principal-Software-Engineer.md [hunk 6]
```diff
@@ -10,27 +10,33 @@
...
 
 <details>
 <summary>Examples</summary>
 
 - I successfully completed complex tasks spanning multiple domains and teams with high impact. 
-- I helped a team reduce complexity through sound architectural thinking.
-- I identified a problem that was timely, important, and impactful, and worked closely with product, engineering, and executive stakeholders to articulate and refine a solution.
-- I handled multiple conflicting priorities and sequenced work resulting in the best collective outcome.
+- I helped a team reduce complexity and risk through sound architectural thinking.
+- I developed a plan to evolve the architecture of one of the core components of Octopus within my impact radius.
```
Comment (zentron) [inline Software-Engineering/L5-Principal-Software-Engineer.md:25]: Key part of this is `within my impact radius`  👍  .

One problem I found previously was being expected to develop and lead improvements to core architecture but being linked to a team which didn't necessitate many "core architectural" changes.  

I.e. The difference between a platform team vs product team are going to have much different impacts in this space.

**Review by zentron — APPROVED** [review]
👍 LGTM. I think this caters for slightly broader impact that might be relevant to experiences in different teams.

File: Software-Engineering/L5-Principal-Software-Engineer.md [hunk 7]
```diff
@@ -39,18 +45,22 @@
 <summary>Examples</summary>
 
 - I was consistently in demand for design and code review.
-- I was recognized by engineers in multiple teams as an expert mentor and teacher.
+- I am actively mentoring multiple Octopus engineers.
+- Engineers at Octopus seek out to work on my teams to learn from me.
+- After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager to uplift their skills in the identified growth area.
```
[the remark this answers is not in the document]
Comment (pawelpabich) [reply Software-Engineering/L5-Principal-Software-Engineer.md:50]: This one seems to be very prescriptive and might not scale.  E.g. I usually work with L4 running the project/team to ensure the uplift happens. L4s should already have a great communication channel established with their manager. 

A more generic version:

` After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager/lead to uplift their skills in the identified growth area.`

File: Software-Engineering/L5-Principal-Software-Engineer.md [hunk 8]
```diff
@@ -10,27 +10,33 @@
...
 
 - I successfully completed complex tasks spanning multiple domains and teams with high impact. 
-- I helped a team reduce complexity through sound architectural thinking.
-- I identified a problem that was timely, important, and impactful, and worked closely with product, engineering, and executive stakeholders to articulate and refine a solution.
-- I handled multiple conflicting priorities and sequenced work resulting in the best collective outcome.
+- I helped a team reduce complexity and risk through sound architectural thinking.
+- I developed a plan to evolve the architecture of one of the core components of Octopus within my impact radius.
+- I identified a problem that was timely, important, and impactful, and worked closely with product, engineering, and executive stakeholders to articulate and refine a solution and delivery plan.
+- I worked with teams to manage multiple conflicting priorities and helped sequence work resulting in the best collective outcome.
+- I collaborated with another principal engineer to ensure that a feature I was helping a team with would support a feature I knew another team was planning.
```
Comment (pawelpabich) [inline Software-Engineering/L5-Principal-Software-Engineer.md:28]: 🤔 I'm not sure we can dictate the scope of projects. Would it be enough for us to connect the relevant teams so they can make the best decision?

**Review by pawelpabich — APPROVED** [review]
LGTM. I find the new examples very useful.

File: Software-Engineering/L5-Principal-Software-Engineer.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (andrewabest) [reply Software-Engineering/L5-Principal-Software-Engineer.md:50]: I'll update this with "lead" as well 👍

File: Software-Engineering/L5-Principal-Software-Engineer.md (hunk 8 above)
Comment (andrewabest) [reply Software-Engineering/L5-Principal-Software-Engineer.md:28]: I don't think we can dictate it, but I think we can influence it from time to time, and that influence can be important and timely.

File: Software-Engineering/L5-Principal-Software-Engineer.md (hunk 7 above)
Comment (andrewabest) [inline Software-Engineering/L5-Principal-Software-Engineer.md:50]:
```suggestion
- After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager or team leader to uplift their skills in the identified growth area.
```

**Pull request merged**